### PR TITLE
refactored redis component

### PR DESCRIPTION
**Pull request by Ma27**
- added third party redis package in order to enhance configuration
- added "keep_bc" parameter in order to keep the behavior of the
  component::redis manifest

### CODE REVIEW

**Comment by Ma27** [comment]
@fadenb @fpletz could you please have a look at that?

File: Puppetfile [hunk 1]
```diff
@@ -16,9 +16,11 @@ mod 'puppetlabs/mysql',      '3.0.0'
...
 mod 'puppetlabs/mongodb',    '0.8.0'
 mod 'puppetlabs/rabbitmq',   '4.1.0'
+mod 'puppetlabs-gcc',        '0.3.0'
 mod 'jfryman/nginx',         '0.2.1'
 mod 'mayflower/php',         '3.4.2'
 mod 'mayflower/hhvm',        '0.1.2'
 mod 'willdurand/nodejs',     '1.8.3'
 mod 'maestrodev/wget',       '1.5.6'
 mod 'nanliu/staging',        '1.0.2'
+mod 'thomasvandoren/redis',  '0.10.0'
```
Comment (fpletz) [inline Puppetfile]: This module compiles redis from source, that's why the gcc module is needed. We prefer installing software via packages.

Why is this needed?

File: site/component/manifests/redis.pp [hunk 2]
```diff
@@ -1,3 +1,15 @@
-class component::redis {
-  package { 'redis-server': }
+class component::redis($keep_bc = true, $instances = {}) {
+  validate_bool($keep_bc)
+  validate_hash($instances)
+
+  if $keep_bc {
+    package { 'redis-server': }
```
Comment (fpletz) [inline site/component/manifests/redis.pp]: Why is backwards compatibility needed? Nothing should depend on the package itself.

Comment (Ma27) [inline site/component/manifests/redis.pp]: this behavior is the actual behavior of the component::redis manifest

File: Puppetfile (hunk 1 above)
Comment (Ma27) [inline Puppetfile]: this module makes it easier to have multiple redis instances, but the gcc dependency is indeed a disadvantage. I need to find another module that provides that feature...

File: site/component/manifests/redis.pp (hunk 2 above)
Comment (fpletz) [inline site/component/manifests/redis.pp]: Yes, but why do you need the package `redis-server` for backwards-compatibility? Nothing should depend on the package itself. Instead, a working redis should be the only thing required. So that's fine. :)

**Comment by Ma27** [comment]
closing, as I don't need that anymore
